extensions/smallrye-context-propagation/runtime/src/main/java/io/quarkus/smallrye/context/runtime/SmallRyeContextPropagationProvider.java

```diff
@@ -1,53 +1,21 @@
 package io.quarkus.smallrye.context.runtime;
 
-import java.util.List;
-import java.util.concurrent.ExecutorService;
-
-import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.Dependent;
 import javax.enterprise.inject.Produces;
-import javax.enterprise.inject.Typed;
 import javax.inject.Singleton;
 
-import org.eclipse.microprofile.context.ManagedExecutor;
 import org.eclipse.microprofile.context.ThreadContext;
 
 import io.quarkus.arc.DefaultBean;
-import io.smallrye.context.SmallRyeManagedExecutor;
-import io.smallrye.context.SmallRyeThreadContext;
 
-@ApplicationScoped
+@Dependent
```

Comment:
What's the rationale for this change? (asking as I would like to understand)

Reply:
The producer is `@Singleton`, i.e. will only be called once and the result is stored in the singleton context. Therefore, there is no need to keep the `SmallRyeContextPropagationProvider` instance around. Not to mention the generated client proxy class.